covalent_ssh_plugin/ssh.py

```diff
@@ -422,3 +309,90 @@ def get_status(self, info_dict: dict) -> Result:
+            args: Positional arguments to be passed to the function.
+            kwargs: Keyword argument to be passed to the function.
+        
+        Returns:
+            An awaitable coroutine which once awaited will return the result
+            of the executed function.
+        """
+
+        operation_id = f"{self.dispatch_id}_{self.node_id}"
+
+        exception = None
+
+        ssh_success, conn = await self._client_connect()
+
+        if not ssh_success:
+            message = f"Could not connect to host '{self.hostname}' as user '{self.username}'"
+            return self._on_ssh_fail(function, args, kwargs, message)
+
+        message = f"Executing node {self.node_id} on host {self.hostname}."
+        app_log.debug(message)
+
+        if self.python3_path == "":
+            cmd = "which python3"
+            # client_in, client_out, client_err = self.client.exec_command(cmd)
+            result = await conn.run(cmd)
+            client_out = result.stdout
+            client_err = result.stderr
+            # self.python3_path = client_out.read().decode("utf8").strip()
+            self.python3_path = client_out.strip()
+
+        if self.python3_path == "":
+            message = f"No Python 3 installation found on host machine {self.hostname}"
+            return self._on_ssh_fail(function, args, kwargs, message)
+
+        cmd = f"mkdir -p {self.remote_cache_dir}"
+
+        result = await conn.run(cmd)
+        client_out = result.stdout
+        client_err = result.stderr
+
+        err = client_err
+        if err != "":
+            app_log.warning(err)
+
+        # Pickle and save location of the function and its arguments:
+        function_file, script_file, remote_function_file, remote_script_file, remote_result_file = self._write_function_files(operation_id, function, args, kwargs)
+
+        await asyncssh.scp(function_file, (conn, remote_function_file))
+        await asyncssh.scp(script_file, (conn, remote_script_file))
+
+        # Run the function:
+        cmd = f"{self.python3_path} {remote_script_file}"
+        result = await conn.run(cmd)
+        err = result.stderr.strip()
+        if err != "":
+            app_log.warning(err)
+
+        # Check that a result file was produced:
+        cmd = f"ls {remote_result_file}"
+        result = await conn.run(cmd)
+        client_out = result.stdout
+        if client_out.strip() != remote_result_file:
+            message = f"Result file {remote_result_file} on remote host {self.hostname} was not found"
+            return self._on_ssh_fail(function, args, kwargs, message)
+
+        # scp the pickled result to the local machine here:
+        result_file = os.path.join(self.cache_dir, f"result_{operation_id}.pkl")
+        await asyncssh.scp((conn, remote_result_file), result_file)
+
+        # Load the result file:
+        with open(result_file, "rb") as f_in:
+            result, exception = pickle.load(f_in)
```

Comment:
Let's switch this plugin to use `json.load` in a followup PR. I get nervous when I see `pickle.load` by the server.

Reply:
Agreed.